esp-hal/src/spi/master.rs

```diff
@@ -2900,6 +2905,15 @@ impl Driver {
         unsafe { &*self.info.register_block }
     }
 
+    fn abort_transfer(&self) {
+        // This method came later than DmaDriver::abort_transfer. That function works
+        // for DMA so I have left it unchanged, but does not work for
+        // CPU-controlled transfers on the ESP32.
+        self.regs().slave().modify(|_, w| w.mode().set_bit());
+        self.regs().slave().modify(|_, w| w.mode().clear_bit());
```

Comment:
Curious to know how you discovered that this does the trick.
Also curious why you only need one update rather than two (one per `modify`).

Reply:
An idea born out of desperation, really.

> Also curious why you only need one update rather than two (one per modify).

I believe we need none, currently, but we will need two as you noticed, if we start supporting other clock sources.